builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -143,7 +150,14 @@ func resourceAwsS3BucketObjectPut(d *schema.ResourceData, meta interface{}) erro
 		putInput.ContentDisposition = aws.String(v.(string))
 	}
 
-	resp, err := s3conn.PutObject(putInput)
```

Comment:
Is there any reason why not use `SSEKMSKeyId` here, instead of the low-level `PutObjectRequest` with headers?

https://github.com/aws/aws-sdk-go/blob/master/service/s3/api.go#L5299

also I think we're going to have to specify `ServerSideEncryption`: https://github.com/aws/aws-sdk-go/blob/master/service/s3/api.go#L5303